Rename `parent_id` to `parent_location_id`

esoergel opened the pull request:
Unfortunately django reserved `sql_location.parent_id` for the primary key of the parent location, so I'd like to switch everything to `parent_location_id`, so couch and SQL are closer.
(wait on tests)

esoergel committed:
Add parent_location_id

Unfortunately django reserved `sql_location.parent_id` for the primary
key of the parent location, so I'd like to switch everything to
`parent_location_id`

esoergel committed:
parent_id -> parent_location_id

esoergel commented:
@czue

czue commented:
looks good but tests didn't run. 

looks like `parent_id` is very common in the codebase. one thought if you're not sure you caught all the references would be to keep the old function as a passthrough and then `soft_assert` yourself if it gets called. only necessary if you're not confident you caught them all though.

esoergel committed:
Add soft assert for deprecated parent_id property

esoergel commented:
I had left `parent_id` in as an alias just in case, but you're correct that that means I'd never get any more confidence about a proper delete.  I've switched it to a property and added in a soft assert.

esoergel commented on corehq/apps/locations/models.py:
```diff
@@ -17,12 +17,19 @@
+def notify_of_deprecation(msg):
+    _assert = soft_assert(notify_admins=True, fail_if_debug=True)
+    message = "Deprecated Locations feature used: {}".format(msg)
+    _assert(False, message)
```
I've got a few things on the chopping block, so I figured I'd make this generic enough to use also in those other places when their time comes

czue commented:
tests caught something:
```
======================================================================
FAIL: test_change_parent (corehq.apps.locations.tests.test_location_import.LocationImportTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/locations/tests/test_location_import.py", line 181, in test_change_parent
    result = import_location(self.domain.name, 'outlet', data)
  File "/mnt/commcare-hq-ro/corehq/apps/locations/tests/test_location_import.py", line 15, in import_location
    return importer.import_location(loc_type, data)
  File "/mnt/commcare-hq-ro/corehq/apps/locations/bulk.py", line 177, in import_location
    consumption,
  File "/mnt/commcare-hq-ro/corehq/apps/locations/bulk.py", line 230, in submit_form
    if self.no_changes_needed(existing, form_data, consumption):
  File "/mnt/commcare-hq-ro/corehq/apps/locations/bulk.py", line 209, in no_changes_needed
    if getattr(existing, key, None) != val:
  File "/mnt/commcare-hq-ro/corehq/apps/locations/models.py", line 825, in parent_id
    notify_of_deprecation("parent_id should be replaced by parent_location_id")
  File "/mnt/commcare-hq-ro/corehq/apps/locations/models.py", line 30, in notify_of_deprecation
    _assert(False, message)
  File "/mnt/commcare-hq-ro/corehq/util/soft_assert/core.py", line 33, in __call__
    return self._call(assertion, msg, obj)
  File "/mnt/commcare-hq-ro/corehq/util/soft_assert/core.py", line 47, in _call
    raise AssertionError(msg)
AssertionError: Deprecated Locations feature used: parent_id should be replaced by parent_location_id
```

czue commented:
:+1: on :white_check_mark: from me

esoergel committed:
Map parent_id to parent_location_id

I looked in to changing the form field to parent_location_id instead,
but that felt like a very large change, as custom code deals with that
form too, and I didn't want to couple that to this PR.

esoergel committed:
Two more old references

esoergel committed:
Merge branch 'master' of github.com:dimagi/commcare-hq into parent-location-id

esoergel commented:
This should be g2g now
@sravfeyn new code buddy

esoergel commented:
@benrudolph

benrudolph commented:
nice
